crates/interpreter/src/exec.rs

```diff
@@ -765,28 +766,43 @@ impl<'m> Thread<'m> {
             Unreachable => return Err(trap()),
             Nop => (),
             Block(b) => self.push_label(self.blocktype(inst, &b), LabelKind::Block),
-            Loop(b) => self.push_label(self.blocktype(inst, &b), LabelKind::Loop(saved)),
+            Loop(b) => {
+                let side_table = self.frame().side_table;
+                self.push_label(
+                    self.blocktype(inst, &b),
+                    LabelKind::Loop(LoopState { parser_data: saved, side_table }),
+                )
+            }
             If(b) => match self.pop_value().unwrap_i32() {
                 0 => {
-                    self.skip_to_else(inst);
+                    self.take_jump(0);
                     self.push_label(self.blocktype(inst, &b), LabelKind::Block);
                 }
-                _ => self.push_label(self.blocktype(inst, &b), LabelKind::If),
+                _ => {
+                    self.frame().skip_jump();
+                    self.push_label(self.blocktype(inst, &b), LabelKind::If);
+                }
             },
             Else => {
-                self.skip_to_end(inst, 0);
+                self.take_jump(0);
                 return Ok(self.exit_label());
             }
-            End => return Ok(self.exit_label()),
-            Br(l) => return Ok(self.pop_label(inst, l)),
+            End => {
+                return Ok(self.exit_label());
```

Comment:
By removing the semi-colon we can get it on the same line as the pattern, as it was before.
```suggestion
                return Ok(self.exit_label())
```